source/common/ssl/utility.cc

```diff
@@ -8,6 +8,15 @@
 namespace Envoy {
 namespace Ssl {
 
+inline ASN1_TIME epochASN1_Time() {
```

Comment:
no inline and make this `const ASN1_TIME&` should make ASAN happy.

Reply:
Looks like it is still not happy :-(

Reply:
Sorry CONSTRUCT_ON_FIRST_USE contains C++ `new` so it doesn't work here.

```C++
const ASN1_TIME& epochASN1_Time() {
  static ASN1_TIME* e = []() -> ASN1_TIME* {
    ASN1_TIME* epoch = ASN1_TIME_new();
    time_t epoch_time = 0;
    ASN1_TIME_set(epoch, epoch_time);
    return epoch;
  }();

  return *e;                                                                                                                                                                                                      }
```
would work.